src/mococrw/pkcs11_engine.h

```diff
@@ -0,0 +1,48 @@
+/*
+ * #%L
+ * %%
+ * Copyright (C) 2018 BMW Car IT GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+#pragma once
+
+#include "openssl_wrap.h"
+#include "mococrw/asymmetric_crypto_ctx.h"
+
+namespace mococrw
+{
+
+/**
+ * A PKCS-11 Engine
+ *
+ */
+class Pkcs11Engine
+{
+public:
+    explicit Pkcs11Engine(const std::string &module_path, const std::string &pin);
+    ENGINE *internal();
+    const ENGINE *internal() const;
+
+    /**
+     * Load public key via Engine.
+     */
+    AsymmetricPublicKey loadPublicKey(const std::string& key_id);
```

Comment:
I'm not sure if should have this functionality here. The other loaders are in the copy of the key objects. I think I would prefer to have them there as well, or let's say at least an option to use it from there so that you can say something like
`AsymmetricPublicKey::readKeyFromEngine(engine&, key_id&)`